obvs/patchscope.py

```diff
@@ -227,20 +248,69 @@ def target_forward_pass(self) -> None:
             self.manipulate_target()
 
     def manipulate_target(self) -> None:
-        (
-            getattr(getattr(self.target_model, self.MODEL_TARGET), self.LAYER_TARGET)[
-                self.target.layer
-            ].output[0][:, self._target_position, :]
-        ) = self._mapped_hidden_state
+
+        # get the specified layer
+        layer = getattr(getattr(self.target_model, self.MODEL_TARGET), self.LAYER_TARGET)[
+            self.target.layer
+        ]
+
+        # if a head index is given, need to access the ATTN and HEAD components
+        if self.target.head is not None:
+            attn = getattr(layer, self.ATTN_TARGET)
+            # TODO may not be .input for other models
+            concat_head_act = getattr(attn, self.HEAD_TARGET).input[0][0]
+
+            # need to reshape the output of head into the specific heads
+            split_head_act = einops.rearrange(
+                concat_head_act, 'batch pos (n_head d_head) -> batch pos n_head d_head',
+                n_head=attn.num_heads, d_head=attn.head_dim
+            )
+
+            # check if the dimensions of the mapped_hidden_state and target head activations match
+            target_act = split_head_act[:, self._target_position, self.target.head, :]
+            if self._mapped_hidden_state.shape != target_act.shape:
+                logger.error('Cannot set activation of head %s in target model with shape'
+                             ' %s to patched activation of source model with shape %s!',
+                             self.target.head, list(target_act.shape),
+                             list(self._mapped_hidden_state.shape))
+                return
```

Comment:
Should we throw an exception here?

Reply:
Yes, good idea. Otherwise the patchscope will continue without patching anything and this might confuse the user.